File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2021-11-01/examples-js/Topics/SBTopicAuthorizationRuleRegenerateKey.js
const { ServiceBusManagementClient } = require("@azure/arm-servicebus");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Regenerates primary or secondary connection strings for the topic.
 *
 * @summary Regenerates primary or secondary connection strings for the topic.
 * x-ms-original-file: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2021-11-01/examples/Topics/SBTopicAuthorizationRuleRegenerateKey.json
 */
async function topicAuthorizationRuleRegenerateKey() {
  const subscriptionId = "e2f361f0-3b27-4503-a9cc-21cfba380093";
  const resourceGroupName = "Default-ServiceBus-WestUS";
  const namespaceName = "sdk-Namespace8408";
  const topicName = "sdk-Topics2075";
  const authorizationRuleName = "sdk-Authrules5067";
  const parameters = { keyType: "PrimaryKey" };
  const credential = new DefaultAzureCredential();
  const client = new ServiceBusManagementClient(credential, subscriptionId);
  const result = await client.topics.regenerateKeys(
    resourceGroupName,
    namespaceName,
    topicName,
    authorizationRuleName,
    parameters
  );
  console.log(result);
}

topicAuthorizationRuleRegenerateKey().catch(console.error);

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2021-11-01/examples-java/Topics/SBTopicAuthorizationRuleRegenerateKey.java

import com.azure.resourcemanager.servicebus.models.KeyType;
import com.azure.resourcemanager.servicebus.models.RegenerateAccessKeyParameters;

/**
 * Samples for Topics RegenerateKeys.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2021-11-01/examples/Topics/
     * SBTopicAuthorizationRuleRegenerateKey.json
     */
    /**
     * Sample code: TopicAuthorizationRuleRegenerateKey.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void topicAuthorizationRuleRegenerateKey(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.serviceBusNamespaces().manager().serviceClient().getTopics().regenerateKeysWithResponse(
            "Default-ServiceBus-WestUS", "sdk-Namespace8408", "sdk-Topics2075", "sdk-Authrules5067",
            new RegenerateAccessKeyParameters().withKeyType(KeyType.PRIMARY_KEY), com.azure.core.util.Context.NONE);
    }
}
